h2o-py/h2o/explanation/_explain.py

```diff
@@ -1305,35 +1308,59 @@ def ice_plot(
             factor_map = _factor_mapper(NumpyFrame(frame[column]).from_factor_to_num(column))
 
         deciles = [int(round(frame.nrow * dec / 10)) for dec in range(11)]
+        deciles[10] = frame.nrow - 1
         colors = plt.get_cmap(colormap, 11)(list(range(11)))
         plt.figure(figsize=figsize)
         for i, index in enumerate(deciles):
+            percentile_string = "{}th Percentile".format(i * 10)
             tmp = NumpyFrame(
                 model.partial_plot(
                     frame,
                     cols=[column],
                     plot=False,
                     row_index=index,
                     targets=target,
-                    nbins=20 if not is_factor else 1 + frame[column].nlevels()[0]
+                    nbins=100 if not is_factor else 1 + frame[column].nlevels()[0],
+                    include_na=True
                 )[0]
             )
+            user_splits = dict()
+            orig_value = frame.as_data_frame()[column][index]
             encoded_col = tmp.columns[0]
+            if _isnan(orig_value):
+                id = np.where(tmp[encoded_col] == tmp.from_factor_to_num(encoded_col)['.missing(NA)'])[0][0] if is_factor else np.where(np.isnan(tmp[encoded_col]))[0][0]
+                orig_null_value = tmp.from_num_to_factor(encoded_col)[tmp[encoded_col][id]] if is_factor else tmp[encoded_col][id]
+                interval = "[{}, {}]".format(orig_null_value,  tmp["mean_response"][id])
+                warnings.warn("Original observation of \"" + encoded_col + "\" for " + percentile_string + " is " + interval + ". Ploting of NAs is not yet supported." )
+            else:
+                user_splits[column] = [orig_value]
```

Comment:
QQ: What's the purpose of this `user_splits` variable? It seems it's always the same dict, just the `orig_value` changes , is that right?

Reply:
this is explained in 1st point of answer to 1st comment about line `pythonnp.where(tmp[encoded_col] == tmp.from_factor_to_num(encoded_col)['.missing(NA)'])[0][0]`